src/test/scala/strawman/collection/test/Test.scala

```diff
@@ -469,6 +469,8 @@ class StrawmanTest {
     val xs9t: mutable.HashMap[Int, String] = xs9
     val xs10 = xs1 ++ xs2
     val xs11: immutable.SortedMap[String, Int] = xs10
+    val xs12 = xs11.collect({ case (k, v) if k.length == v => (v, k) }: scala.PartialFunction[(String, Int), (Int, String)])
```

Comment:
The type annotation (`scala.PartialFunction[…]`) is required with dotty, Scala 2.12 and Scala 2.13.